File: benchmark/output/lua-love/openai/breakout_game.lua
local paddle_x = 360
local paddle_y = 570
local paddle_width = 80
local paddle_height = 12
local paddle_speed = 300

local ball_x = 400
local ball_y = 400
local ball_radius = 6
local ball_vx = 200
local ball_vy = -200

local bricks = {}
local brick_width = 90
local brick_height = 20
local brick_alive = {1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0}

local score = 0
local game_over = false

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("Breakout")

    for i = 0, 7 do
        table.insert(bricks, {x = i * 100 + 5, y = 50})
    end
end

function love.update(dt)
    if game_over then return end

    -- Paddle movement
    if love.keyboard.isDown("left") then
        paddle_x = paddle_x - paddle_speed * dt
    end
    if love.keyboard.isDown("right") then
        paddle_x = paddle_x + paddle_speed * dt
    end

    -- Ball movement
    ball_x = ball_x + ball_vx * dt
    ball_y = ball_y + ball_vy * dt

    -- Ball collision with walls
    if ball_x < ball_radius or ball_x > 800 - ball_radius then
        ball_vx = -ball_vx
    end
    if ball_y < ball_radius then
        ball_vy = -ball_vy
    end

    -- Ball collision with paddle
    if ball_y >= paddle_y - ball_radius and ball_x >= paddle_x and ball_x <= paddle_x + paddle_width then
        ball_vy = -ball_vy
    end

    -- Ball collision with bricks
    for i, brick in ipairs(bricks) do
        if brick_alive[i] == 1.0 then
            if ball_x > brick.x and ball_x < brick.x + brick_width and ball_y > brick.y and ball_y < brick.y + brick_height then
                brick_alive[i] = 0.0
                ball_vy = -ball_vy
                score = score + 1
            end
        end
    end

    -- Check for game over
    if ball_y > 600 then
        game_over = true
    end
end

function love.draw()
    -- Draw paddle
    love.graphics.rectangle("fill", paddle_x, paddle_y, paddle_width, paddle_height)

    -- Draw ball
    love.graphics.circle("fill", ball_x, ball_y, ball_radius)

    -- Draw bricks
    for i, brick in ipairs(bricks) do
        if brick_alive[i] == 1.0 then
            love.graphics.rectangle("fill", brick.x, brick.y, brick_width, brick_height)
        end
    end

    -- Draw score
    love.graphics.print("Score: " .. score, 10, 10)

    -- Draw game over
    if game_over then
        love.graphics.print("GAME OVER", 350, 300)
    end
end

File: vibe-lang/benchmark/output/python-pygame/openai/breakout_game.py
import pygame
import sys

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("Breakout")
clock = pygame.time.Clock()

# Paddle
paddle_x = 360.0
paddle_speed = 300.0

# Ball
ball_x, ball_y = 400.0, 400.0
ball_radius = 6
ball_vx, ball_vy = 200.0, -200.0

# Bricks
brick_width, brick_height = 90, 20
brick_alive = [1.0] * 8
brick_positions = [(i * 100 + 5, 50) for i in range(8)]

# Game state
score = 0
game_over = False

running = True
while running:
    dt = clock.tick(60) / 1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

    keys = pygame.key.get_pressed()
    if keys[pygame.K_LEFT]:
        paddle_x -= paddle_speed * dt
    if keys[pygame.K_RIGHT]:
        paddle_x += paddle_speed * dt

    # Keep paddle within screen bounds
    paddle_x = max(0, min(paddle_x, 720))

    # Update ball position
    ball_x += ball_vx * dt
    ball_y += ball_vy * dt

    # Ball collision with walls
    if ball_x < ball_radius or ball_x > 800 - ball_radius:
        ball_vx *= -1
    if ball_y < ball_radius:
        ball_vy *= -1

    # Ball collision with paddle
    if ball_y >= 564 and paddle_x <= ball_x <= paddle_x + 80:
        ball_vy *= -1

    # Ball collision with bricks
    for i, (bx, by) in enumerate(brick_positions):
        if brick_alive[i] == 1.0:
            if bx <= ball_x <= bx + brick_width and by <= ball_y <= by + brick_height:
                brick_alive[i] = 0.0
                ball_vy *= -1
                score += 1

    # Check if ball is out of bounds
    if ball_y > 600:
        game_over = True

    # Drawing
    screen.fill((0, 0, 0))
    pygame.draw.rect(screen, (255, 255, 255), (int(paddle_x), 570, 80, 12))
    pygame.draw.circle(screen, (255, 255, 255), (int(ball_x), int(ball_y)), ball_radius)

    for i, (bx, by) in enumerate(brick_positions):
        if brick_alive[i] == 1.0:
            pygame.draw.rect(screen, (255, 255, 255), (bx, by, brick_width, brick_height))

    font = pygame.font.Font(None, 36)
    score_text = font.render(f"Score: {score}", True, (255, 255, 255))
    screen.blit(score_text, (10, 10))

    if game_over:
        game_over_text = font.render("GAME OVER", True, (255, 0, 0))
        screen.blit(game_over_text, (350, 300))

    pygame.display.flip()

pygame.quit()
sys.exit()